Added author & title to post revisions

mike182uk opened the pull request:
no issue

Added `created_by` and `title` columns to `post_revisions` table

mike182uk committed:
Added author to post revisions

no issue

Added created_by column to post_revisions table

mike182uk commented on ghost/core/core/server/data/migrations/versions/5.45/2023-04-17-11-05-add-post-revision-author.js:
```diff
@@ -0,0 +1,7 @@
+module.exports = createAddColumnMigration('post_revisions', 'created_by', {
+    type: 'string',
+    maxlength: 24,
+    nullable: true
```
Set this to `nullable` to not interfere with existing records

allouis replied:
👍

mike182uk committed:
Added title to post revisions

no issue

Added title column to post_revisions table

allouis commented on ghost/core/core/server/data/migrations/versions/5.45/2023-04-17-11-05-add-post-revision-author.js:
```diff
@@ -0,0 +1,7 @@
+const {createAddColumnMigration} = require('../../utils');
+
+module.exports = createAddColumnMigration('post_revisions', 'created_by', {
```
Appreciate you following existing convention, but I think because these are not resources we should use `author_id` instead!

allouis replied:
cc @daniellockyer wdyt?

mike182uk replied:
@allouis No problemo, thats updated 👍

allouis reviewed (commented):
Just one comment about column name otherwise seems 👍

mike182uk committed:
Changed created_by to author_id

no issue

Changed the name of the created_by column to author_id in the post revisions table

mike182uk committed:
Added fk for post revisions author id

no issue

Added foreign key for post revisions author id to refernce users
